pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -495,23 +564,38 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		if identityMatches(set, replicas[i]) && storageMatches(set, replicas[i]) && retentionMatch {
 			continue
 		}
-		// Make a deep copy so we don't mutate the shared cache
-		replica := replicas[i].DeepCopy()
-		if err := ssc.podControl.UpdateStatefulPod(ctx, updateSet, replica); err != nil {
-			return &status, err
-		}
+		actions = append(actions, func() (bool, error) {
+			// Make a deep copy so we don't mutate the shared cache
+			replica := replicas[i].DeepCopy()
+			if err := ssc.podControl.UpdateStatefulPod(ctx, updateSet, replica); err != nil {
+				return true, err
+			}
+			return false, nil
+		})
 	}
 
+	if monotonic {
+		for _, fn := range actions {
+			if shouldExit, err := fn(); shouldExit || err != nil {
+				return &status, err
+			}
+		}
```

Comment:
You may want to clear actions, otherwise if we have some actions that don't return shouldexit=true we may execute them multiple times (e.g. createMissingPersistentVolumeClaims).

Reply:
Done